code/datums/gas_mixture.dm

```diff
@@ -61,901 +57,927 @@ What are the archived variables for?
-		var/self_temperature_delta = 0
+	temperature += self_temperature_delta
 
-		if(abs(delta_temperature) > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
-			var/self_heat_capacity = heat_capacity_archived()
+	return 1
+	//Logic integrated from: temperature_mimic(model, conduction_coefficient) for efficiency
 
-			if((model.heat_capacity > MINIMUM_HEAT_CAPACITY) && (self_heat_capacity > MINIMUM_HEAT_CAPACITY))
-				var/heat = conduction_coefficient*delta_temperature* \
-					(self_heat_capacity*model.heat_capacity/(self_heat_capacity+model.heat_capacity))
+/datum/gas_mixture/temperature_share(datum/gas_mixture/sharer, conduction_coefficient)
 
-				self_temperature_delta = -heat/(self_heat_capacity*group_multiplier)
+	var/delta_temperature = (temperature_archived - sharer.temperature_archived)
+	if(abs(delta_temperature) > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
+		var/self_heat_capacity = heat_capacity_archived()
+		var/sharer_heat_capacity = sharer.heat_capacity_archived()
 
-		if((abs(self_temperature_delta) > MINIMUM_TEMPERATURE_DELTA_TO_SUSPEND) \
-			&& (abs(self_temperature_delta) > MINIMUM_TEMPERATURE_RATIO_TO_SUSPEND*temperature_archived))
-			return 0
+		if((sharer_heat_capacity > MINIMUM_HEAT_CAPACITY) && (self_heat_capacity > MINIMUM_HEAT_CAPACITY))
+			var/heat = conduction_coefficient*delta_temperature* \
+				(self_heat_capacity*sharer_heat_capacity/(self_heat_capacity+sharer_heat_capacity))
 
-		temperature += self_temperature_delta
+			temperature -= heat/self_heat_capacity
+			sharer.temperature += heat/sharer_heat_capacity
 
-		return 1
-		//Logic integrated from: temperature_mimic(model, conduction_coefficient) for efficiency
+/datum/gas_mixture/temperature_mimic(turf/model, conduction_coefficient, border_multiplier)
+	var/delta_temperature = (temperature - model.temperature)
+	if(abs(delta_temperature) > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
+		var/self_heat_capacity = heat_capacity()//_archived()
 
-	temperature_share(datum/gas_mixture/sharer, conduction_coefficient)
+		if((model.heat_capacity > MINIMUM_HEAT_CAPACITY) && (self_heat_capacity > MINIMUM_HEAT_CAPACITY))
+			var/heat = conduction_coefficient*delta_temperature* \
+				(self_heat_capacity*model.heat_capacity/(self_heat_capacity+model.heat_capacity))
+
+			if(border_multiplier)
+				temperature -= heat*border_multiplier/self_heat_capacity
+			else
+				temperature -= heat/self_heat_capacity
 
-		var/delta_temperature = (temperature_archived - sharer.temperature_archived)
-		if(abs(delta_temperature) > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
-			var/self_heat_capacity = heat_capacity_archived()
-			var/sharer_heat_capacity = sharer.heat_capacity_archived()
+/datum/gas_mixture/temperature_turf_share(turf/simulated/sharer, conduction_coefficient)
+	var/delta_temperature = (temperature_archived - sharer.temperature)
+	if(abs(delta_temperature) > MINIMUM_TEMPERATURE_DELTA_TO_CONSIDER)
+		var/self_heat_capacity = heat_capacity()
 
-			if((sharer_heat_capacity > MINIMUM_H
```

Comment:
Also contains an equation from human breathing that I made into a generic proc (literally above your comment) and a proc I made to reverse that equation, plus proofs.